### PR TITLE
[CALCITE-6374] LatticeSuggester throw NullPointerException when agg c…

### PR DESCRIPTION
**Pull request by YiwenWu**
…all covered with cast
https://issues.apache.org/jira/browse/CALCITE-6374

Ensure that the `name` field of `MutableMeasure` is not empty. If `name` is empty, an alias will be automatically generated.

### CODE REVIEW

File: core/src/test/java/org/apache/calcite/materialize/LatticeSuggesterTest.java [hunk 1]
```diff
@@ -721,6 +721,36 @@ private void checkDerivedColumn(Lattice lattice, List<String> tables,
     checkFoodmartSimpleJoin(CalciteAssert.SchemaSpec.FAKE_FOODMART);
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-6374">[CALCITE-6374]
+   * LatticeSuggester throw NullPointerException when agg call covered with cast </a>. */
+  @Test void testCastExpression() throws Exception {
```
Comment (caicancai) [inline core/src/test/java/org/apache/calcite/materialize/LatticeSuggesterTest.java:727]: Maybe the name testCastExpression is not clear enough. Is there a clearer way to use it?

Comment (YiwenWu) [reply core/src/test/java/org/apache/calcite/materialize/LatticeSuggesterTest.java]: Adjusted test method name.

**Review by caicancai — APPROVED** [review]
LGTM

**Review by macroguo-ghy — APPROVED** [review]
looks good, please squash your commits

**Comment by YiwenWu** [comment]
@macroguo-ghy thank you for the review, I have squashed commits.

**Pull request merged**